Makefile

```diff
@@ -0,0 +1,40 @@
+LANG		:= en
+LANG_FILE	:= $(shell test -f LANGS.md.bak && echo LANGS.md.bak || echo LANGS.md)
+LANG_DATA	:= $(shell grep "$(LANG)/" $(LANG_FILE))
+LANG_NAME	:= $(shell echo "$(LANG_DATA)" | sed 's/.*\[\(.*\)\].*/\1/')
+
+help:
+	@echo
+	@echo "Usage: make command [LANG=<language_short_code>]"
+	@echo
+	@echo "Language short codes are found in LANGS.md. ( For eg: LANG=es )"
+	@echo "Default Language is 'en'(English)."
+	@echo "Commands:"
+	@echo "  init	- Initialise and Start the server."
+	@echo "  serve	- Start honkit server locally for development."
+	@echo "  setup	- Setup the temporary language for the development."
+	@echo "  reset	- Reset the temporarily set language."
+	@echo "  check	- Check for root directory for various dependencies."
+	@echo "  help	- Display command list."
+
+check: book.json LANGS.md package.json $(shell which node) $(shell which npx)
+
+reset: 
+	@if test -f "LANGS.md.bak"; then \
+		mv -f LANGS.md.bak LANGS.md && echo "Language file is reset"; \
+	else \
+		echo "Language file backup is not found."; \
+	fi
+
+setup: reset
+ifneq ($(LANG_FILE), "LANGS.md.bak")
+	@cp LANGS.md LANGS.md.bak && echo "Language file backup is created."
+	@echo "$(LANG_DATA)" > LANGS.md && echo "Language list is reduced to $(LANG_NAME) for development"
+endif
+
+serve:
+	@npx honkit serve || (npm install && npx honkit serve)
```

Comment:
Shouldn't you create some rule like:

```make
node_modules:
 	@npm install

serve: node_modules check
 	@npx honkit serve
```

Reply:
Lets take the scenario where npm modules are already installed. npm will have to run again before start serving when `make serve` is ran. But it is possible since it live reload server. Personally, I dont prefer this method incase I have network problem and my local server will not running. This is waste of time. But if your method is more acceptable generally, i can change that. Let me know your thoughts on my point of view.